### PR TITLE
Builder integration tests for Dockerfile outside context as a unit test

### PR DESCRIPTION
**Pull request by tkopczynski**
Reimplementing 2 integration tests for Dockerfile outside the context as a unit test.

Part of resolving #20784

Signed-off-by: Tomasz Kopczynski tomek@kopczynski.net.pl

### CODE REVIEW

File: builder/dockerfile/internals_test.go [hunk 1]
```diff
@@ -53,3 +53,50 @@ func TestEmptyDockerfile(t *testing.T) {
...
+
+	b := &Builder{options: options, context: context}
+
+	err = b.readDockerfile()
+
+	if err == nil {
+		t.Fatalf("No error when executing test for Dockerfile outside the build context")
+	}
+
+	expectedError := "Forbidden path outside the build context: ../../Dockerfile"
```
Comment (justincormack) [inline builder/dockerfile/internals_test.go]: Given that the previous code had different cases for Windows paths, would it be more reliable to just test for `"Forbidden path outside the build context:"`? (I know the test passed, but...)

**Comment by tkopczynski** [comment]
hey @justincormack I was not sure about this myself, so I changed this as you suggested

**Comment by LK4D4** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐮

**Pull request merged**